lib/Dialect/SV/SVOps.cpp

```diff
@@ -1055,6 +1057,130 @@ LogicalResult PAssignOp::verify() {
   return success();
 }
 
+namespace {
+// This represents a slice of an array.
+struct ArraySlice {
+  Value array;
+  APInt start, end; // Represent a range [satrt, end).
```

Comment:
Can we represent `start` as a value and the `size` as an `APInt`? This would simplify `getArraySlice` and instead `isMergableSlice` could do the work of extracting the constant (or proving that some dynamic values are consecutive).

Reply:
Changed `start` to a value and `size` to uint64_t. `ArraySlice` is basically same as `HWOps.cpp/mergeConcatSlices/Slice` so it would be nicer if we can unify them in the future (also possibly solved by introducing Vector dialect).